server/src/main/java/org/elasticsearch/index/engine/ElasticsearchConcurrentMergeScheduler.java

```diff
@@ -78,6 +79,17 @@ protected boolean verbose() {
         return super.verbose();
     }
 
+    // TODO: this is temporarily, remove this override and enable multithreaded merges for all kind of merges
+    @Override
+    public Executor getIntraMergeExecutor(MergePolicy.OneMerge merge) {
+        // Enable multithreaded merges only for force merge operations
+        if (merge.getStoreMergeInfo().mergeMaxNumSegments != -1) {
+            return super.getIntraMergeExecutor(merge);
+        } else {
+            return null;
+        }
+    }
+
```

Comment:
I do not think this should return `null`. It should instead return a same thread executor.